api/v1alpha1/storageconsumer_types.go

```diff
@@ -37,12 +37,37 @@ const (
 )
 
 // StorageConsumerSpec defines the desired state of StorageConsumer
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.RadosNamespace) || has(self.RadosNamespace)", message="Value is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.SubVolumeGroup) || has(self.SubVolumeGroup)", message="Value is required once set"
 type StorageConsumerSpec struct {
 	// Enable flag ignores a reconcile if set to false
 	Enable bool `json:"enable,omitempty"`
 	// StorageQuotaInGiB describes quota for the consumer
 	// +optional
 	StorageQuotaInGiB int `json:"storageQuotaInGiB,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +kubebuilder:validation:MaxLength=512
+	RadosNamespace string `json:"radosNamespace"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +kubebuilder:validation:MaxLength=512
+	SubVolumeGroup             string                         `json:"subVolumeGroup"`
```

Comment:
The validation here does not make sense. This need to be configurable all the way until the Consumer is enabled. At that point we should freeze it. 

But I question the need of adding these fields as part of this PR. Why do we need to couple these changes with the rest of the changes in this PR?

Reply:
removed.